src/main/java/org/jabref/cli/ArgumentProcessor.java

```diff
@@ -239,6 +253,114 @@ public boolean hasParserResults() {
         return loaded;
     }
 
+    private void writeXMPtoPdf(List<ParserResult> loaded, String filesAndCitekeys, JabRefPreferences preferences) {
+
+        ParserResult pr = loaded.get(loaded.size() - 1);
+        BibDatabaseContext databaseContext = pr.getDatabaseContext();
+        BibDatabase dataBase = pr.getDatabase();
+
+        XmpPdfExporter xmpPdfExporter = new XmpPdfExporter(preferences.getXmpPreferences());
+
+        if (filesAndCitekeys.equals("all")) {
+            for (BibEntry entry : dataBase.getEntries()) {
+                writeXMPtoPDFsOfEntry(databaseContext, dataBase, entry, preferences, xmpPdfExporter);
+            }
+            return;
+        }
+
+        Vector<String> citeKeys = new Vector<>();
+        Vector<String> pdfs = new Vector<>();
+        for (String fileOrCiteKey : filesAndCitekeys.split(",")) {
+            if (fileOrCiteKey.toLowerCase(Locale.ROOT).endsWith(".pdf")) {
+                pdfs.add(fileOrCiteKey);
+            } else {
+                citeKeys.add(fileOrCiteKey);
+            }
+        }
+
+        writeXMPtoPdfByCitekey(databaseContext, dataBase, citeKeys, preferences, xmpPdfExporter);
+        writeXMPtoPdfByFileNames(databaseContext, dataBase, pdfs, preferences, xmpPdfExporter);
+
+    }
+
+    private void writeXMPtoPDFsOfEntry(BibDatabaseContext databaseContext, BibDatabase dataBase, BibEntry entry, JabRefPreferences preferences, XmpPdfExporter xmpPdfExporter) {
+        boolean writtenToAFile = false;
+
+        for (LinkedFile file : entry.getFiles()) {
+            if (file.getFileType().toLowerCase(Locale.ROOT).equals("pdf")) {
+                Optional<Path> filePath = file.findIn(databaseContext, preferences.getFilePreferences());
+                if (filePath.isEmpty()) {
+                    System.err.println(Localization.lang("Skipped - PDF %0 does not exist", filePath.get()));
```

Comment:
Is the logger not available in CLI?

Reply:
It is, but many other options use System.err so I tried to keep it the same.
Will change everything to Logger.error if that is the more appropriate way of printing errors.

Reply:
Let's talk about this with @koppor , since refactoring the logging is not the focus of your work, so I would suggest to postpone it.

Reply:
With everything I only meant in my code. Will do that for now and we can discuss about refactoring the rest.